pixeldata/src/lib.rs

```diff
@@ -0,0 +1,468 @@
+        let bits_stored = get_bits_stored(self)?;
+        let high_bit = get_high_bit(self)?;
+        let pixel_representation = get_pixel_representation(self)?;
+
+        let decoded_pixel_data = match pixel_data.value() {
+            Value::PixelSequence {
+                fragments,
+                offset_table: _,
+            } => {
+                if fragments.len() > 1 {
+                    // Bundle fragments and decode multi-frame dicoms
+                    NotSupported {
+                        message: "Multi-frame dicoms are not yet supported",
+                    }
+                    .fail()?
+                }
+                let decoded_frame = decode_single_frame_compressed(
+                    &fragments[0],
+                    cols.into(),
+                    rows.into(),
+                    pi_type,
+                    ts_type,
+                    samples_per_pixel,
+                    bits_allocated,
+                    bits_stored,
+                    high_bit,
+                    pixel_representation,
+                );
+                decoded_frame.unwrap().to_vec()
+            }
+            Value::Primitive(p) => {
+                // Non-encoded, just return the pixel data
+                p.to_bytes().to_vec()
+            }
+            Value::Sequence { items: _, size: _ } => InvalidPixelData.fail()?,
+        };
+
+        return Ok(DecodedPixelData {
+            data: decoded_pixel_data,
+            cols: cols.into(),
+            rows: rows.into(),
+            endianness,
+            photometric_interpretation,
+            samples_per_pixel,
+            bits_allocated,
+            bits_stored,
+            high_bit,
+            pixel_representation,
+        });
+    }
+}
+
+/// Get the Columns of the dicom
+fn get_cols(obj: &DefaultDicomObject) -> Result<u16> {
+    Ok(obj
+        .element_by_name("Columns")
+        .context(MissingRequiredField)?
+        .uint16()
+        .context(CastValueError)?)
+}
+
+/// Get the Rows of the dicom
+fn get_rows(obj: &DefaultDicomObject) -> Result<u16> {
+    Ok(obj
+        .element_by_name("Rows")
+        .context(MissingRequiredField)?
+        .uint16()
+        .context(CastValueError)?)
+}
+
+/// Get the PhotoMetricInterpretation of the Dicom
+fn get_photometric_interpretation(obj: &DefaultDicomObject) -> Result<String> {
+    Ok(obj
+        .element_by_name("PhotometricInterpretation")
+        .context(MissingRequiredField)?
+        .string()
+        .context(CastValueError)?
+        .trim()
+        .to_string())
+}
+
+/// Get the SamplesPerPixel of the Dicom
+fn get_samples_per_pixel(obj: &DefaultDicomObject) -> Result<u16> {
+    Ok(obj
+        .element_by_name("SamplesPerPixel")
+        .context(MissingRequiredField)?
+        .uint16()
+        .context(CastValueError)?)
+}
+
+/// Get the BitsAllocated of the Dicom
+fn get_bits_allocated(obj: &DefaultDicomObject) -> Result<u16> {
```

Comment:
The `get_` prefix is not very idiomatic in Rust. https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter. This one for example could be just `bits_allocated`.